tests/Unit/Domain/Test_Domain.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+#include <memory>
+
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/Domain.hpp"
+#include "Utilities/Gsl.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+template <size_t Dim>
+void test_domain_construction(
+    const Domain<Dim, Frame::Grid>& domain,
+    const std::vector<std::unordered_map<Direction<Dim>, BlockNeighbor<Dim>>>&
+        expected_block_neighbors,
+    const std::vector<std::unordered_set<Direction<Dim>>>&
+        expected_external_boundaries) {
+  const auto& blocks = domain.blocks();
```

Comment:
Check blocks, expected_block_neighbors and expected_external_boundaries are the same size